internal/repos/other.go

```diff
@@ -71,6 +72,12 @@ func NewOtherSource(ctx context.Context, svc *types.ExternalService, cf *httpcli
 		return nil, err
 	}
 
+	if envvar.SourcegraphDotComMode() {
+		if c.DotComPublic {
+			svc.Unrestricted = true
+		}
+	}
```

Comment:
Could this be further simplified?

```suggestion
	if envvar.SourcegraphDotComMode() && c.DotComPublic {
		svc.Unrestricted = true
	}
```

Reply:
lol yes, idk why I nested it like that. Thanks!